### PR TITLE
Fix rendering of controller.ingress.path

### PR DESCRIPTION
**Pull request by hazzik**
# What this PR does / why we need it

Fixes rendering `controller.ingress.path` options. It does not make sense to put it under the branch of when `paths` are explicitly defined, it should be in the block where `paths` are empty.

# Special notes for your reviewer

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/templates/jenkins-controller-ingress.yaml [hunk 1]
```diff
@@ -42,11 +42,11 @@ spec:
           serviceName: {{ template "jenkins.fullname" . }}
           servicePort: {{ .Values.controller.servicePort }}
 {{ end }}
-{{- else }}
-{{ tpl (toYaml .Values.controller.ingress.paths | indent 6) . }}
 {{- if .Values.controller.ingress.path }}
         path: {{ .Values.controller.ingress.path }}
 {{- end -}}
+{{- else }}
+{{ tpl (toYaml .Values.controller.ingress.paths | indent 6) . }}
```
Comment (timja) [inline charts/jenkins/templates/jenkins-controller-ingress.yaml:49]: can you add a test for this?
https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/tests/jenkins-controller-ingress-test.yaml

Comment (hazzik) [reply charts/jenkins/templates/jenkins-controller-ingress.yaml:49]: Done. Not sure if I have done it correctly though.

**Comment by timja** [comment]
needs a changelog update to fix conflicts

**Comment by hazzik** [comment]
@timja done

**Review by timja — APPROVED** [review]

**Pull request merged**